kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/security/oauth/KopOAuthBearerSaslServer.java

```diff
@@ -111,6 +111,9 @@ public Object getNegotiatedProperty(String propName) {
             return tokenForNegotiatedProperty.authDataSource();
         }
         if (USER_NAME_PROP.equals(propName)) {
+            if (tokenForNegotiatedProperty.tenant() != null) {
```

Comment:
For the OAUTH Bearer mechanism, isn't the `AUTH_DATA_SOURCE_PROP` used to get the tenant information from the auth details? I thought `USER_NAME_PROP` is used for PLAIN authentication. I am not sure, but can you please check and confirm this? I think that is the reason for the test case failure where it was getting `NULL` instead of the supplied tenant name.

Reply:
The `AuthDataSource` don't have tenant information, so here we use `USER_NAME_PROP ` to pass the tenant information to `SaslAuthenticator`.

The failed test is caused by wrong delimiter. See: https://github.com/streamnative/kop/pull/1728/commits/a93313b9e6a9ffc7ba3c04dc64ad5ed495c14870

Reply:
Ohh okay.